src/v6tools/Subnet.php

```diff
@@ -62,4 +62,41 @@ public function isInSubnet($ipv6addr) {
         }
         return true;
     }
+
+    /**
+     * Generates EUI64 address
+     *
+     * @param string Mac address
+     */
+    public function getEUI64Address($mac) {
+        // Validate prefix length
+        if ($this->preflen != 64) {
+            throw new \InvalidArgumentException("Prefix length is not 64.");
```

Comment:
Can this error message by a bit more descriptive? It's clear for people who know the standard that autoconf is only valid in /64. I think it would be helpful for people using this library to have this context in the error message, e.g.

"Generating of IP addresses using EUI64 is only allowed in autoconfigured networks (/64 subnets). You must use a 64 bit prefix".

Reply:
Not a problem at all.